src/kbmod/work_unit.py

```diff
@@ -541,6 +548,237 @@ def to_fits(self, filename, overwrite=False):
+            (see below) as well as the per_image_wcs information for
+            the whole set. This will have the given filename.
+            -One image fits file containing all of the image data for
+            every LayeredImage in the ImageStack. This will have the
+            image index infront of the given filename, e.g.
+            "0_filename.fits".
+
+
+        Primary File:
+            0 - Primary header with overall metadata
+            1 or "metadata" - The data provenance metadata
+            2 or "kbmod_config" - The search parameters
+        Individual Image File:
+            Image extensions for the science layer ("SCI_i"),
+            variance layer ("VAR_i"), mask layer ("MSK_i"), and
+            PSF ("PSF_i") of each image.
+
+        Parameters
+        ----------
+        filename : `str`
+            The base filename to which to write the data.
+        directory: `str`
+            The directory to place all of the FITS files.
+            Recommended that you have one directory per
+            sharded file to avoid confusion.
+        overwrite : bool
+            Indicates whether to overwrite an existing file.
+        """
+        logger.info(
+            f"Writing WorkUnit shards with {self.im_stack.img_count()} images with main file {filename} in {directory }"
+        )
+        primary_file = os.path.join(directory, filename)
+        if Path(primary_file).is_file() and not overwrite:
+            raise FileExistsError(f"WorkUnit file {filename} already exists.")
+
+        # Set up the initial HDU list, including the primary header
+        # the metadata (empty), and the configuration.
+        hdul = fits.HDUList()
+        pri = fits.PrimaryHDU()
+        pri.header["NUMIMG"] = self.im_stack.img_count()
+        pri.header["REPRJCTD"] = self.reprojected
+        pri.header["HELIO"] = self.heliocentric_distance
+        for i in range(len(self.constituent_images)):
+            pri.header[f"GEO_{i}"] = self.geocentric_distances[i]
+
+        # If the global WCS exists, append the corresponding keys.
+        if self.wcs is not None:
+            append_wcs_to_hdu_header(self.wcs, pri.header)
+
+        n_constituents = len(self.constituent_images)
+        pri.header["NCON"] = n_constituents
+
+        hdul.append(pri)
+
+        meta_hdu = fits.BinTableHDU()
+        meta_hdu.name = "metadata"
+        hdul.append(meta_hdu)
+
+        config_hdu = self.config.to_hdu()
+        config_hdu.name = "kbmod_config"
+        hdul.append(config_hdu)
+
+        for i in range(self.im_stack.img_count()):
+            layered = self.im_stack.get_single_image(i)
+            c_indices = self._per_image_indices[i]
+            n_indices = len(c_indices)
+            sub_hdul = fits.HDUList()
+
+            img_wcs = self.get_wcs(i)
+            sci_hdu = raw_image_to_hdu(layered.get_science(), img_wcs)
+            sci_hdu.name = f"SCI_{i}"
+            sci_hdu.header["NIND"] = n_indices
```

Comment:
Not for this PR, but a thought for the future: If we are going to use the `NIND` and `IND_{j}` header values a lot, it might be worth breaking these into a helper function. Something like "fits_header_read_image_indices()" and "fits_header_append_image_indices()"